packages/ui/src/bounty/components/BountyPreviewHeader/BountyPreviewHeader.tsx

```diff
@@ -149,24 +177,17 @@ const getReward = (entry: WorkEntry) => {
   return isBountyEntryStatusWinner(entry.status) ? new BN(entry.status.reward) : undefined
 }
 
-const SuccessfulStageButtons = React.memo(({ bounty, activeMember, t }: BountyHeaderButtonsProps) => {
-  const userEntry = useMemo(() => bounty.entries?.find((entry) => entry.worker.id === activeMember?.id), [bounty])
+const SuccessfulStageButtons = React.memo(({ bounty, t, userEntry, isContributor }: BountyHeaderButtonsProps) => {
   const entryId = userEntry?.id
   const reward = userEntry ? getReward(userEntry) : undefined
-  const { winner, passed } =
-    useMemo(() => bounty.entries?.find((entry) => entry.worker.id === activeMember?.id), [bounty]) || {}
-  const winnerConditions = winner && entryId && reward
-  const isContributor = useMemo(
-    () => bounty.contributors?.some((contributor) => contributor.actor?.id === activeMember?.id),
-    [bounty]
-  )
+  const winnerConditions = userEntry?.winner && entryId && reward
   return (
     <>
       <ButtonGhost size="large">
         <BellIcon /> {t('common:buttons.notifyAboutChanges')}
       </ButtonGhost>
       {winnerConditions && <ClaimRewardButton bountyId={bounty.id} entryId={entryId} reward={reward} />}
-      {(passed || isContributor) && isContributor ? (
+      {(userEntry?.passed || isContributor) && isContributor ? (
         <WithdrawContributionButton bounty={bounty} />
       ) : (
```

Comment:
I don't think `<WithdrawContributionButton />` should show at all during the successful stage